src/Core/System/StateMachine/Util/StateMachineGraphvizDumper.php

```diff
@@ -171,8 +171,8 @@ private function addOptions(array $options): string
     {
         $code = [];
         foreach ($options as $k => $v) {
-            \assert(\is_string($k));
-            \assert(\is_string($v));
+            \assert($k === null || is_scalar($k) || $k instanceof \Stringable);
```

Comment:
I don't think this is correct for the key? can keys even be null?

Reply:
php arrays can be null

Reply:
weird, so they can. Not correct here tho imo.

Reply:
they are not really `null` then, are they? :thinking: aren't they implicitly cast to an empty string?

Reply:
yeah: https://3v4l.org/sMZjS#v8.3.11

Reply:
good point